### PR TITLE
Updates igxTreeGrid and igxHierarchicalGrid Templates to Support Right Pinning

### PR DESCRIPTION
**Pull request by 3phase**
Related issue #6676   

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-row.component.html [hunk 1]
```diff
@@ -121,6 +121,64 @@
     </ng-template>
     <ng-container *ngTemplateOutlet="col.visibleIndex === 0 ? treeCellTemplate : cellTemplate"></ng-container>
 </ng-template>
+<ng-container *ngIf="pinnedColumns.length > 0 && !grid.isPinningToStart">
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-row.component.html]: Can't these be extrated in a template that can be used for both right and left pinning?

Comment (3phase) [reply projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-row.component.html]: Sure, it's done now.

**Comment by MayaKirova** [comment]
@3phase 
For the HierarchicalGrid the horizontal scrollbar does not start/end at the correct position:
image

**Comment by 3phase** [comment]
> @3phase
> For the HierarchicalGrid the horizontal scrollbar does not start/end at the correct position if pinning is set to end:
> image

It's fixed now, thanks for the remark

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 2]
```diff
@@ -5961,4 +5961,18 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
             advancedFilteringDialog.closeDialog();
         }
     }
+
+    /**
+     * @hidden
+     * @internal
+     */
+    public get getScrollbarSize() {
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts]: This is named as a method but is actually a getter. Can be named more descriptively - pinStartSize or similar.
Also check can be simplified to isPinningToStart ? pinnedWidth : headerFeaturesWidth

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.html [hunk 3]
```diff
@@ -137,11 +148,14 @@
...
-    <div class="igx-grid__scroll-start" [style.width.px]='pinnedWidth' [hidden]="pinnedWidth === 0"></div>
-    <div class="igx-grid__scroll-main" [style.width.px]='unpinnedWidth'>
+    <div class="igx-grid__scroll-start" 
+        [style.width.px]='pinnedWidth ? getScrollbarSize : (featureColumnsWidth() ? featureColumnsWidth() : 0)'
+        [hidden]="!pinnedWidth ? (!featureColumnsWidth() ? true : false) : false"></div>
+    <div class="igx-grid__scroll-main" [style.width.px]='unpinnedWidth - featureColumnsWidth()'>
         <ng-template igxGridFor [igxGridForOf]='[]' #scrollContainer>
         </ng-template>
     </div>
+    <div class="igx-grid__scroll-end" [style.width.px]='pinnedWidth - featureColumnsWidth()' *ngIf="!isPinningToStart"></div>
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.html]: No need to subtract the feature's columns. This breaks the end scrollbar size:
image

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.html [hunk 4]
```diff
@@ -121,7 +127,7 @@
 </div>
 
 <div class="igx-grid__scroll" [style.height]="'18px'" #scr [hidden]="isHorizontalScrollHidden">
-    <div class="igx-grid__scroll-start" [style.width.px]='pinnedWidth' [hidden]="pinnedWidth === 0"></div>
+    <div class="igx-grid__scroll-start" [style.width.px]='getScrollbarSize' [hidden]="pinnedWidth === 0"></div>
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.html]: There is no end pin section. One should be added, same as in the base grid.

**Pull request merged**